### PR TITLE
Upgrade influxdb client to v2

### LINKED ISSUE #11: Monitoring > Upgrade the Influxdb Client
# Problem

The influxdb client is being deprecated which causing monitoring doesn't work well with latest dependencies.

> NOTE: The Go client library now has a "v2" version, with the old version being deprecated. The new version can be imported at import "github.com/influxdata/influxdb/client/v2". It is not backwards-compatible.

https://github.com/influxdata/influxdb/tree/master/client/

# Solution

Upgrade the influxdb client to v2. https://github.com/influxdata/influxdb/tree/master/client/v2 to fix this.

### PR DESCRIPTION
**Pull request by junhuif**
Close #11.

### CODE REVIEW

**Comment by junhuif** [comment]
Didn't write unit test for this, but tested it manually. For fyi:

```sh
// Empty Config
➜  go-playground go run main.go
2018/05/18 14:29:52 influxdb monitoring url  not absolute url
exit status 1

// No Password
➜  go-playground go run main.go
2018/05/18 12:01:45 influxdb monitoring url https://aigle_dev@influxdb.theplant-dev.com:8086/aigle_dev not password
exit status 1

// No Database
➜  go-playground go run main.go
2018/05/18 12:01:51 influxdb monitoring url https://aigle_dev:PASSWORD@influxdb.theplant-dev.com:8086 not database
exit status 1
```

```sh
// Worked
➜  go-playground go run main.go
level=info ts=2018-05-18T12:02:28.062421232+08:00 caller=influxdb.go:92 scheme=https username=aigle_dev database=aigle_dev host=influxdb.theplant-dev.com:8086 msg="influxdb instrumentation writing to https://aigle_dev@influxdb.theplant-dev.com:8086/aigle_dev"
```

File: monitoring/influxdb.go [hunk 1]
```diff
@@ -30,22 +31,46 @@ func NewInfluxdbMonitor(config InfluxMonitorConfig, logger log.Logger) (Monitor,
 		return nil, errors.Wrapf(err, "couldn't parse influxdb url %v", monitorURL)
 	} else if !u.IsAbs() {
 		return nil, errors.Errorf("influxdb monitoring url %v not absolute url", monitorURL)
+	} else if u.Scheme != "http" && u.Scheme != "https" {
```
Comment (bodhi) [inline monitoring/influxdb.go]: This check already exists in the InfluxDB client:

https://github.com/influxdata/influxdb/blob/531dc497177bd423b539db35cc0a48b36c4a280c/client/v2/client.go#L95

File: monitoring/influxdb.go [hunk 2]
```diff
@@ -92,20 +117,37 @@ func (im influxdbMonitor) InsertRecord(measurement string, value interface{}, ta
...
+	pt, err := influxdb.NewPoint(measurement, tags, fields, at)
+
+	if err != nil {
+		im.logger.Error().Log(
+			"err", err,
+			"database", im.database,
+			"measurement", measurement,
+			"value", value,
+			"tags", tags,
+			"during", "influxdb.NewPoint",
```
Comment (bodhi) [inline monitoring/influxdb.go]: Remember `logger.With(...)` for `database`, `measurement`, `during`.

**Comment by junhuif** [comment]
> How much work do you think it would be to convert the monitor to a batching monitor?

Maybe couple hours (< 1 day). (Have a talk with Azuma to get a basic of idea for doing this kind of things, thanks @azumads).

> At the moment, each data point is sent in a single request, right? 

Right.

**Comment by junhuif** [comment]
> How much work do you think it would be to convert the monitor to a batching monitor?

@bodhi Moved to #13. For fyi, I won't do this directly as it may costing > 1 day, so if you think it's oaky to do that right now, just let me know.

**Comment by bodhi** [comment]
since you have it fresh in your mind, can you work on it now please?

**Comment by junhuif** [comment]
> since you have it fresh in your mind, can you work on it now please?

Sorry, I found that I missed this message in that day. So I didn't do #13 yet. (Happy to do that when I'm free)

**Pull request merged**